src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -429,6 +536,30 @@ private JsonRpcError ReadError(JToken json)
             };
         }
 
+        private void ClearProgressObject(JToken requestId)
+        {
+            try
+            {
+                long reqId = requestId.Value<long>();
+
+                lock (this.progressMap)
+                {
+                    long progressId;
+
+                    if (this.requestProgressMap.TryGetValue(reqId, out progressId))
+                    {
+                        this.requestProgressMap.Remove(reqId);
+                        this.progressMap.Remove(progressId);
+                    }
+                }
+            }
+            catch (FormatException e)
+            {
+                this.Rpc.TraceSource.TraceData(TraceEventType.Error, (int)JsonRpc.TraceEvents.RequestIdCastError, e);
+                throw e;
```

Comment:
I don't think we need to catch and trace here with a special TraceEvents enum value. The exception itself should describe what went wrong, and our caller should trace out all errors the same. So many things can go wrong with processing a message and a special ID for each individual one I don't think would serve us well, and we'd surely miss some even if we tried. I suggest you remove the catch block entirely.
Also: never `throw e;` inside a catch block. That replaces the callstack on the exception with a new one as if it were originally thrown from here, so you lose evidence of the original failure. Instead if you were to rethrow from a catch block, just write `throw;`.